### PR TITLE
docs(website): fix heading level inconsistencies

### PR DESCRIPTION
**Pull request by Josh-Cena**
## PR Checklist

-   [ ] Addresses an existing issue: fixes #0
-   [ ] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [ ] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

Make the TOC more consistent and easier to navigate

### CODE REVIEW

**Comment by Josh-Cena** [comment]
This looks quite problematic (also in other PRs). I can't see why the build fails

**Review by armano2 — APPROVED** [review]

File: docs/development/architecture/ASTS.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ title: ASTs
 sidebar_label: ASTs
 ---
 
-## Abstract Syntax Trees (AST)s
+# Abstract Syntax Trees (AST)s
```
Comment (JoshuaKGoldberg) [inline docs/development/architecture/ASTS.md:7]: @Josh-Cena was this intentional? The lint complaint is correct:

```
docs/development/architecture/ASTS.md:7 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Abstract Syntax Trees (AST)s"]
```

Comment (Josh-Cena) [reply docs/development/architecture/ASTS.md:7]: Ah wait, I didn't start the site locally... Forgot to hide the front matter title

**Review by bradzacher — APPROVED** [review]

**Comment by bradzacher** [comment]
no need to bother with merging main into your branch - we can keep track of what gets merged and merge for you if it's required before landing the PR :)

**Comment by Josh-Cena** [comment]
Yeah, just getting bored & hitting the "update" button🙈 Hope it doesn't get confusing

**Comment by Josh-Cena** [comment]
Meanwhile I still have no clue about the failing Netlify build...

**Comment by bradzacher** [comment]
> Meanwhile I still have no clue about the failing Netlify build...

That's on our end.. we need to fix our build step.
I believe it's something like we're not building the entire project and instead building a smaller part of it? I don't know exactly as I haven't seen the build config!

**Pull request merged**